sqlglot/dialects/hive.py

```diff
@@ -209,6 +212,23 @@ class Hive(Dialect):
     # https://spark.apache.org/docs/latest/sql-ref-identifier.html#description
     NORMALIZATION_STRATEGY = NormalizationStrategy.CASE_INSENSITIVE
 
+    ANNOTATORS = {
+        **Dialect.ANNOTATORS,
+        exp.If: lambda self, e: self._annotate_by_args(e, "true", "false", promote=True),
+        exp.Coalesce: lambda self, e: self._annotate_by_args(
+            e, "this", "expressions", promote=True
+        ),
+    }
+
+    # Support only the non-ANSI mode (default for Hive, Spark2, Spark)
+    COERCES_TO = defaultdict(set, deepcopy(TypeAnnotator.COERCES_TO))
```

Comment:
We don't need `deepcopy` for this, you only mutate the dict's keys, so a shallow copy is fine.
```suggestion
    COERCES_TO = defaultdict(set, TypeAnnotator.COERCES_TO.copy())
```